static_src/test/unit/stores/activity_store.spec.js

```diff
@@ -0,0 +1,139 @@
+    ActivityStore._data = Immutable.List();
+    ActivityStore._fetching = false;
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('constructor()', function() {
+    it('should start data as empty array', function() {
+      expect(ActivityStore.getAll()).toBeEmptyArray();
+    });
+  });
+
+  describe('on event fetch', function () {
+    it('should call cfApi.fetchSpaceEvents', function () {
+      var spy = sandbox.spy(cfApi, 'fetchSpaceEvents');
+
+      AppDispatcher.handleViewAction({
+        type: activityActionTypes.EVENTS_FETCH,
+        spaceGuid: 'fakeSpaceGuid'
+      });
+
+      expect(spy).toHaveBeenCalledOnce();
+    });
+  });
+
+  describe('on events received', function () {
+    it('should merge activity and emit change', function () {
+      const activity = [
+        {
+          guid: 'fakeActivityGuidOne',
+          name: 'fakeActivityNameOne',
+          activity_type : 'event'
+        },
+        {
+          guid: 'fakeActivityGuidTwo',
+          name: 'fakeActivityNameTwo',
+          activity_type : 'event'
+        }
+      ];
+      const spy = sandbox.spy(ActivityStore, 'emitChange');
+
+      AppDispatcher.handleServerAction({
+        type: activityActionTypes.EVENTS_RECEIVED,
+        events: wrapInRes(activity)
+      });
+
+
+      expect(ActivityStore.getAll()).toEqual(activity);
+      expect(spy).toHaveBeenCalledOnce();
+    });
+  });
+
+  describe('on logs fetch', function () {
+    it('should call cfApi.fetchAppLogs', function () {
+      var spy = sandbox.spy(cfApi, 'fetchAppLogs');
+
+      AppDispatcher.handleViewAction({
+        type: activityActionTypes.LOGS_FETCH,
+        appGuid: 'fakeAppGuid'
+      });
+
+      expect(spy).toHaveBeenCalledOnce();
+    });
+  });
+
+  describe('on logs received', function () {
+    it('should merge activity and emit change', function () {
+      const logs = [
+        {
+          message: "federalist.18f.gov - [17/08/2016:00:09:56.467 +0000] \"GET /socket.io/?__sails_io_sdk_version=0.11.0&__sails_io_sdk_platform=browser&__sails_io_sdk_language=javascript&EIO=3&transport=polling&t=1471392596201-8120 HTTP/1.1\" 200 0 90 \"https://federalist.18f.gov/\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36\" 127.0.0.1:34350 x_forwarded_for:\"50.0.192.84\" x_forwarded_proto:\"https\" vcap_request_id:5e288c8d-d9d8-4589-7956-0427ab11e33f response_time:0.003466191 app_id:32f77e21-d504-4b4a-91c3-ea4c6bcc47e5\n"
+        }
+      ];
+
+      const spy = sandbox.spy(ActivityStore, 'emitChange');
+
+      AppDispatcher.handleServerAction({
+        type: activityActionTypes.LOGS_RECEIVED,
+        logs
+      });
+
+      expect(ActivityStore.getAll().length).toEqual(1);
+      expect(spy).toHaveBeenCalledOnce();
+    });
+  });
+
+  describe('parseLogItem()', function () {
+    it('should parse log items', function () {
```

Comment:
The logs API returns big strings that represent log items (such as the one below), so parsing is using RegExp to parse out the data into named variables

```
  {
    "message": "dashboard.cloud.gov - [19/08/2016:13:56:53.239 +0000] \"GET / HTTP/1.1\" 302 0 310 \"https://login.cloud.gov/\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36\" 127.0.0.1:45574 x_forwarded_for:\"50.139.84.209\" x_forwarded_proto:\"https\" vcap_request_id:ed21da85-42eb-4107-bce0-6805e20b9900 response_time:0.166580506 app_id:ff813a1f-a455-404e-a283-c9addc60ae29\n"
  }
```